core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,33 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
-          // TODO(bhenning): Consolidate the backend rights domain objects and
-          // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
+          // Check if the response from the backend does not contradict
+          // putParams.
+          var isPrivateResponse = (response.data.status === 'private');
+          if (isPrivateResponse === isPublic) {
+            if (isPublic) {
```

Comment:
I don't think these logs makes sense here. I would expect this service simply to identify errors and pass them onto the calling service to digest and output errors, if it wishes to.